gradle/plugins/com.liferay.ide.gradle.core/src/com/liferay/ide/gradle/core/GradleProjectProvider.java

```diff
@@ -82,52 +90,173 @@ public synchronized ILiferayProject provide( Object adaptable )
+
+        if( serviceName != null )
         {
-            retval = ProjectCore.createErrorStatus( e );
+            sb.append( "-s " + serviceName + " " );
         }
 
-        if( retval.isOK() )
+        if( packageName != null )
         {
-            ProjectImportConfiguration configuration = new ProjectImportConfiguration();
-            GradleDistributionWrapper from = GradleDistributionWrapper.from( GradleDistribution.fromBuild() );
-            configuration.setGradleDistribution( from );
-            configuration.setProjectDir( location.toFile() );
-            configuration.setApplyWorkingSets( false );
-            configuration.setWorkingSets( new ArrayList<String>() );
-            new SynchronizeGradleProjectJob(
-                configuration.toFixedAttributes(), configuration.getWorkingSets().getValue(),
-                AsyncHandler.NO_OP ).schedule();
+            sb.append( "-p " + packageName + " " );
+        }
+
+        sb.append( "\"" + projectName + "\" " );
+
+        try
+        {
+            final String[] ret = BladeCLI.execute( sb.toString() );
+
+            final String errors = checkForErrors(ret);
+
+            if( errors.length() > 0 )
+            {
+                retval = GradleCore.createErrorStatus( "Project create error: " + errors );
+                return retval;
+            }
+
+            if( ret.length == 0 )
+            {
+                ElementList<ProjectName> projectNames = op.getProjectNames();
+                
+                projectNames.insert().setName( projectName );
+                
+                IPath projecLocation = location;
+
+                final String lastSegment = location.lastSegment();
+
+                if( location != null && location.segmentCount() > 0 )
+                {
+                    if( !lastSegment.equals( projectName ) )
+                    {
+                        projecLocation = location.append( projectName );
+                    }
+                }
+
+                ProjectImportConfiguration configuration = new ProjectImportConfiguration();
+                GradleDistributionWrapper from = GradleDistributionWrapper.from( GradleDistribution.fromBuild() );
+                configuration.setGradleDistribution( from );
+                configuration.setProjectDir( projecLocation.toFile() );
+                configuration.setApplyWorkingSets( false );
+                configuration.setWorkingSets( new ArrayList<String>() );
+                SynchronizeGradleProjectJob synchronizeGradleProjectJob =
+                    new SynchronizeGradleProjectJob(
+                        configuration.toFixedAttributes(), configuration.getWorkingSets().getValue(),
+                        AsyncHandler.NO_OP );
+
+                final IPath finalClassPath =
+                    getClassFilePath( projectName, className, packageName, projectTemplateName, projecLocation );
+
+                final File finalClassFile = finalClassPath.toFile();
```

Comment:
why this we need to use IFile? This action will be done before the project import to current workspace, so its not a workspace project until now.